core/src/main/resources/reference.conf

```diff
@@ -268,6 +268,21 @@ datastax-java-driver {
     warn-on-init-error = true
   }
 
+  # Whether to schedule reconnection attempts if the driver fails to reach any contact point on the
+  # first initialization attempt.
+  #
+  # If this is true, the driver will retry according to the reconnection policy. The
+  # `SessionBuilder.build()` call -- or the future returned by `SessionBuilder.buildAsync()` --
+  # won't complete until a contact point has been reached.
+  #
+  # If this is false and no contact points are available, the driver will fail with an
+  # AllNodesFailedException.
+  #
+  # Required: yes
+  # Modifiable at runtime: no
+  # Overridable in a profile: no
+  advanced.reconnect-on-init = false
```

Comment:
This is a good idea.  I think in typical case we want to fail fast and users can opt in to this feature if they want.